scripts/apply-rulesets.sh

```
@@ -88,46 +88,49 @@ apply_repo() {
-  [ -n "$target" ] || { echo "::error::usage: $0 --repo owner/repo | <repo-name> [--dry-run] [<name>...]" >&2; return 2; }
+  [[ -d "$RULESETS_DIR" ]] || { echo "::error::rulesets dir not found: $RULESETS_DIR" >&2; return 1; }
+  [[ -n "$target" ]] || { echo "::error::usage: $0 --repo owner/repo | <repo-name> [--dry-run] [<name>...]" >&2; return 2; }
 
   # Select the ruleset files: the named ones, else the fleet allowlist.
   local files=()
-  if [ "${#names[@]}" -eq 0 ]; then names=("${FLEET_RULESETS[@]}"); fi
+  if [[ "${#names[@]}" -eq 0 ]]; then names=("${FLEET_RULESETS[@]}"); fi
   local n
   for n in "${names[@]}"; do
-    [ -f "${RULESETS_DIR}/${n}.json" ] && files+=("${RULESETS_DIR}/${n}.json") \
+    [[ -f "${RULESETS_DIR}/${n}.json" ]] && files+=("${RULESETS_DIR}/${n}.json") \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)